### ISSUE TITLE
Extend chart with descendant generations

### ISSUE THREAD

**Issue by godzil3** [magicsunday/webtrees-fan-chart#95]
Is is possible to deveop funtionality to include descendant generations in space down below the chart? I mean default chart is 210 degree, to full circle left 150 dedgree and for example 140 degree could be used for small fan chart with descendant generations
thanks!

**Comment by godzil3** [comment]
here's quick sketch 
![fan-chart-210-gradient](https://github.com/magicsunday/webtrees-fan-chart/assets/45084859/34c18ee7-048c-4041-a24f-8160c7c480e6)

**Comment by vd-meer-net** [comment]
I didn't know anything about this request, I changed the cod eof this addon to include this information.
Was just looking for a way to get it to the original creator.

You can see my version of the plugin on this site:
[https://familie.vd-meer.net/tree/van-der-meer/webtrees-fan-chart/X1153?generations=3&fanDegree=210&fontScale=100&hideEmptySegments=0&showColorGradients=0&showChildren=1&showParentMarriageDates=0&innerArcs=2](url)

I can't get it nicely in a pull-request.
The modification is based on v2.7.2 of the original. I know there are some problems...:
- Only the male-female-colors work
- No child-data if there are no parents
- Sometimes spacing is a little bit too large
- Translation of the config option is missing

I hope you can include this in your version.

[webtrees-fan-chart-including-children.zip](https://github.com/user-attachments/files/19302193/webtrees-fan-chart-including-children.zip)

**Comment by vd-meer-net** [comment]
Followed the right procedure, I think. So, I forked your version and added all my changes to the code.

You can find it here:
https://github.com/vd-meer-net/webtrees-fan-chart/tree/showChildren

Fixed point 2,4 and 5 before the commit.
Only issues right now are:
 - Only the male-female-colors work, not the color gradient
 - Sometimes spacing is a little bit too large

Let me know what you think

**Comment by vd-meer-net** [comment]
My update is now in both version avaible:
https://github.com/vd-meer-net/webtrees-fan-chart/tree/showChildren

And for 3.0.1:
https://github.com/vd-meer-net/webtrees-fan-chart/tree/3.0.1-showChildren

**Comment by magicsunday** [comment]
v3.3.0 adds descendant support to the fan chart: partners and their children are shown as arcs below the ancestor section. Includes marriage arcs, family colors per partner family, and places when arc width allows. Click navigation and SVG/PNG export work for descendants too.

Planned for upcoming releases:
- Configurable descendant depth (grandchildren, great-grandchildren)
- Images in descendant arcs

Feedback on the current implementation is welcome. Keeping this issue open until the full feature set is delivered.

**Comment by vd-meer-net** [comment]
Thanks for implementing this feature. I was updating my version to your 3.1. But I'll stop my version, yours is much better. Thanks...

**Comment by vd-meer-net** [comment]
One suggestion maybe, I was working on. Give the partners/children a color depending on gender.

**Comment by TheDutchJewel** [comment]
> One suggestion maybe, I was working on. Give the partners/children a color depending on gender.

I agree. By default, visual distinctions are made between the different genders, but not if the “Show family colors” option is checked. If those distinctions are made visible again, the diagram becomes much more informative.

**Comment by TheDutchJewel** [comment]
> v3.3.0 adds descendant support to the fan chart: partners and their children are shown as arcs below the ancestor section.

If "Show descendants" is checked, the children are displayed, but in an illogical chronological order "from right to left" instead of "from left to right". Can this be fixed?

**Comment by TheDutchJewel** [comment]
> Configurable descendant depth (grandchildren, great-grandchildren)

That would be a nice and valuable addition.

**Comment by TheDutchJewel** [comment]
By the way, "Show family colors" only works if "Show parent marriage dates" is also checked.
Why is that, and can this be added to the text under "Show family colors"?

**Comment by magicsunday** [comment]
**v3.3.3 released** — includes fixes related to the descendant feature:

**Done:**
- Children now appear in chronological order (oldest left, youngest right)
- Family colors work correctly regardless of whether marriage dates are shown
- Various fixes for marriage arcs, date formatting, and separator lines in the descendant section

**Planned for future releases:**
- Gender-based colors for descendants (as suggested by @vd-meer-net and @TheDutchJewel)
- Configurable descendant depth (grandchildren, great-grandchildren)

**Comment by magicsunday** [comment]
@TheDutchJewel Regarding "Show family colors only works if Show parent marriage dates is also checked" — I was unable to reproduce this. Family colors work independently of the marriage dates setting in v3.3.3. Could you verify with the latest version? If the issue persists, please open a separate issue with a screenshot so I can investigate further.

**Comment by TheDutchJewel** [comment]
Thanks for fixing the chronological order of children.

> If the issue persists, please open a separate issue with a screenshot so I can investigate further.
See: https://github.com/magicsunday/webtrees-fan-chart/issues/193

**Comment by magicsunday** [comment]
@vd-meer-net @TheDutchJewel — could you clarify what exactly you'd like for the gender colors? I see two possible interpretations:

**Option A** (TheDutchJewel's reading): Keep the family colors as the arc background, but make the thin **color strip at the outer edge** show the gender color even when "Show family colors" is enabled. Currently that strip is grey when family colors are on.

**Option B** (vd-meer-net's reading): The descendant arcs themselves (partner ring + children ring) get a gender-based **background color** instead of (or in addition to) the family color.

Which one matches what you had in mind? Or both?

**Comment by magicsunday** [comment]
After experimenting with **Option B** (gender-colored descendant arcs), I'd like to push back on it: replacing the position-based pastel hues with sex-based hues makes the descendant section visually clash with the ancestor section — siblings of the same sex all share an identical color, and the soft family-grouping effect from the ancestor side is lost. The chart loses its visual cohesion.

So I'd suggest going with **Option A** only: the thin gender-color strip at the outer edge of each arc remains visible even when "Show family colors" is enabled. This adds gender information without disturbing the existing design language. Already implemented in the v3.3.4-dev build — the strip uses the webtrees theme's `--sex-f-bg` / `--sex-m-bg` / `--sex-u-bg` CSS variables, so it adapts to the active webtrees theme.

@vd-meer-net If you still want gender as the primary signal in the descendant arcs, an alternative would be to disable "Show family colors" — then the position-based pastel goes away and the gender strip becomes the dominant color cue. Would that work for you?

**Comment by TheDutchJewel** [comment]
I'd like to see an example to get an idea of what it will look like. Is that possible?

**Comment by magicsunday** [comment]
<img width="1320" height="955" alt="Image" src="https://github.com/user-attachments/assets/6291577c-e7ed-4030-824e-defdb2bda386" />

**Comment by TheDutchJewel** [comment]
Why use soft colors for gender? I think bold colors are much clearer, like the [primer theme](https://github.com/ekdahl/webtrees-primer-theme).

**Comment by magicsunday** [comment]
the strip uses the webtrees theme's --sex-f-bg / --sex-m-bg / --sex-u-bg CSS variables, so it adapts to the active webtrees theme.

**Comment by magicsunday** [comment]
A quick update on the descendant generations part of this issue:

I experimented with extending the chart beyond direct children — adding grandchildren and great-grandchildren as additional outer rings (configurable 1–3 generations). Unfortunately, in real-world trees this turned out to be very hard to render in a visually pleasing way:

- With realistic family sizes (e.g. 10+ children across multiple branches), the grandchild and great-grandchild arcs become so narrow that no reasonable font size fits, and any attempt at uniform font scaling either shrinks readable arcs unnecessarily or leaves the narrow ones unreadable anyway.
- Mixing inner-label (chord) and outer-label (radial) rendering on the same ring depending on arc width produced inconsistent results and made the chart look unbalanced.
- Capping the font per depth or per datum produced visually "backwards" situations where grandchildren ended up with larger text than their own parents.

Given how quickly the lower half of the chart runs out of usable angular space once you go beyond direct children, I've decided to drop the configurable descendant-generations feature for now and keep the descendants section limited to partners + direct children only. The gender-color edge strip from the earlier commit remains.

If anyone has a concrete layout proposal that handles wide family trees gracefully at 2–3 descendant generations, I'm happy to revisit.

**Comment by magicsunday** [comment]
Closing this issue — the descendant part has been addressed as far as layout constraints allow (partners + direct children with optional gender-color edge strip). If someone has a concrete proposal for displaying further descendant generations in a way that works with realistic family sizes, please open a new issue with a layout mockup. That keeps the discussion focused and actionable.
